libvast/include/vast/concept/parseable/vast/pattern.hpp

```diff
@@ -29,11 +29,15 @@ struct access::parser_base<pattern>
 
   template <class Iterator>
   bool parse(Iterator& f, const Iterator& l, pattern& a) const {
-    if (!slash_delimited_string{}(f, l, a.str_)) {
+    auto str = std::string{};
+    if (!slash_delimited_string{}(f, l, str))
       return false;
-    }
-    auto case_insensitive_flag = parsers::chr{pattern::case_insensitive_flag};
-    a.case_insensitive_ = case_insensitive_flag(f, l, unused);
+    auto case_insensitive
+      = parsers::chr{pattern::case_insensitive_flag}(f, l, unused);
```

Comment:
```suggestion
      = parsers::chr{pattern::case_insensitive_flag}(f, l);
```

This *should* work as well.

Reply:
Pretty sure it doesn't, unless you make that `ignore(parsers::chr{...})(f, l)`.

Reply:
I designed all parsers to both take their attribute or `unused`. If not, that's a bug in the framework.

The idea is that you can _always_ use a parser to just validate a grammar. There exists a separate overload because this often requires significantly less work, as no assignment to the attribute needs to take place.

Reply:
Right, but you still need to pass in either the attribute (char in this case) or unused if there is an attribute. Only if all attributes are _unused_ you are allowed to call the parser without a third argument.

Reply:
That would be a bug then. If you go through the top-level `operator()` of the CRTP base, the two-argument form defaults to `unused`. I was not aware that this can cause downstream instantitation issues. I looked at the code and it now looks different from the past compared to, say, this overload:

```cpp
template <size_t N, class Attribute = unused_type>
  bool operator()(const char (&r)[N], Attribute& a = unused) const {
    // Because there exists overload of std::begin / std::end for char arrays,
    // we must have this overload and strip the NUL byte at the end.
    auto f = r;
    auto l = r + N - 1; // No NUL byte.
    return derived().parse(f, l, a) && f == l;
  }
```

So some refactoring has changed a fundamental property of the API UX.

Reply:
I tried removing `unused` and strangely, it "works on my machine". https://github.com/tenzir/vast/pull/2964/commits/b11aa6a32f031b396e0f95a005262c9239ab5fca